beacon-chain/blockchain/core.go

```diff
@@ -459,7 +476,39 @@ func (b *BeaconChain) saveCanonical(block *types.Block) error {
 	if err != nil {
 		return err
 	}
-	return b.db.Put([]byte(canonicalHeadKey), enc)
+	return b.db.Put(CanonicalHeadLookupKey, enc)
+}
+
+// retrieveBlock retrieves a block from the db using its slot number and hash.
+func (b *BeaconChain) retrieveBlock(hash [32]byte) (*types.Block, error) {
```

Comment:
This and `doesBlockExist` are duplicates of `getBlock` and `hasBlock`

Reply:
some methods were duplicated when I merged in your pr, removing them

Reply:
+1 to getBlock

Reply:
Alright will change it